pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -152,17 +154,27 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 func (c *Reconciler) reconcile(ctx context.Context, pr *v1alpha1.PipelineRun) error {
 	p, err := c.pipelineLister.Pipelines(pr.Namespace).Get(pr.Spec.PipelineRef.Name)
```

Comment:
For consistency across taskrun and pipelinerun reconcilers, I would suggest using similar `InitializeConditions()` function at the beginning of `reconcile` function.

Reply:
I can do that - can you explain _why_ we need to do this, I just don't quite understand what it's for.

Reply:
So [TaskRunStatus has `InitializeConditions`](https://github.com/knative/build-pipeline/blob/394a25bed41a584880e09f5a1596e190033f186c/pkg/apis/pipeline/v1alpha1/taskrun_types.go#L97), but I'm still not understanding what ends up calling it - can you explain what I should add, where and how it will work?

(I'm guessing add `InitializeConditions` to PipelineRunStatus - and should I update both controllers to call this at the beginning of the reconcile loop?)

Reply:
I am sorry for doing a crappy job in explaining that. Let me rephrase, `InitializeConditions` function in taskrun checks if ducktype condition is nil and initialize condition defined in `taskrunCondSet` to `Unknown` status. Uses ducktype defined function under the hood.

Ref for function definition: https://github.com/knative/build-pipeline/blob/394a25bed41a584880e09f5a1596e190033f186c/vendor/github.com/knative/pkg/apis/duck/v1alpha1/condition_set.go#L295-L302



> both controllers to call this at the beginning of the reconcile loop?

Yes exactly.

PipelineRun does not have `InitializeConditions`, so you could consider initializing that function which can remove the following block of code.

```
if tr.Status.GetCondition(duckv1alpha1.ConditionSucceeded) == nil {
 tr.Status.SetCondition(&duckv1alpha1.Condition{
 Type:   duckv1alpha1.ConditionSucceeded,
 Status: corev1.ConditionUnknown,
 Reason: ReasonRunning,
 })
 }
```

An example of using `InitializeConditions ` function in serving: 
https://github.com/knative/serving/blob/c3a95a733f195cce81922299059c3acfe27ce30b/pkg/reconciler/v1alpha1/service/service.go#L141

I recommended the change because ducktype considers all these cases and has functions defined already. It would be a good practice to adopt these functions in controllers and not duplicate code.

Reply:
@bobcatfish : If this is increasing the scope of Pr, feel free to discard the recommendation. Functionality is not affected.

Reply:
thanks for the detailed explanation @shashwathi ! I'll give it a shot and if I do something wonky let me know, then we could maybe put it in another PR instead but I'll see if we can do it here.